specs/CustomDownload.md

```diff
@@ -0,0 +1,771 @@
+  /// the following interrupt reasons may automatically resume without you calling any methods:
+  /// `COREWEBVIEW2_DOWNLOAD_INTERRUPT_REASON_SERVER_NO_RANGE`,
+  /// `COREWEBVIEW2_DOWNLOAD_INTERRUPT_REASON_FILE_HASH_MISMATCH`,
+  /// `COREWEBVIEW2_DOWNLOAD_INTERRUPT_REASON_FILE_TOO_SHORT`.
+  [propget] HRESULT CanResume([out, retval] BOOL* canResume);
+}
+```
+## .NET/ WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    runtimeclass CoreWebView2Download;
+    runtimeclass CoreWebView2DownloadStartingEventArgs;
+    runtimeclass CoreWebView2DownloadStateChangedEventArgs;
+
+    enum CoreWebView2DownloadState
+    {
+        InProgress = 0,
+        Interrupted = 1,
+        Completed = 2,
+    };
+    enum CoreWebView2DownloadInterruptReason
+    {
+        None = 0,
+        FileFailed = 1,
+        FileAccessDenied = 2,
+        FileNoSpace = 3,
+        FileNameTooLong = 4,
+        FileTooLarge = 5,
+        FileVirusInfected = 6,
+        FileTransientError = 7,
+        FileBlocked = 8,
+        FileSecurityCheckFailed = 9,
+        FileTooShort = 10,
+        FileHashMismatch = 11,
+        FileSameAsSource = 12,
+        NetworkFailed = 13,
+        NetworkTimeout = 14,
+        NetworkDisconnected = 15,
+        NetworkServerDown = 16,
+        NetworkInvalidRequest = 17,
+        ServerFailed = 18,
+        ServerNoRange = 19,
+        ServerBadContent = 20,
+        ServerUnauthorized = 21,
+        ServerCertProblem = 22,
+        ServerForbidden = 23,
+        ServerUnreachable = 24,
+        ServerContentLengthMismatch = 25,
+        ServerCrossOriginRedirect = 26,
+        UserCanceled = 27,
+        UserShutdown = 28,
+        UserPaused = 29,
+        DownloadProcessCrashed = 30,
+    };
+
+    runtimeclass CoreWebView2Settings
+    {
+        // CoreWebView2Settings
+        // There are other settings that we are not showing
+        Boolean ShouldDisplayDefaultDownloadDialog { get; set; };
+    }
+
+    runtimeclass CoreWebView2DownloadStateChangedEventArgs
+    {
+        // CoreWebView2DownloadStateChangedEventArgs
+        CoreWebView2DownloadInterruptReason InterruptReason { get; };
+    }
+
+    runtimeclass CoreWebView2DownloadStartingEventArgs
+    {
+        // CoreWebView2DownloadStartingEventArgs
+        CoreWebView2Download Download { get; };
+
+        Boolean Cancel { get; set; };
+
+        String ResultFilePath { get; set; };
+
+        Windows.Foundation.Deferral GetDeferral();
+    }
+
+    runtimeclass CoreWebView2
+    {
+        // There are other API in this interface that we are not showing
+        event Windows.Foundation.TypedEventHandler<CoreWebView2, CoreWebView2DownloadStartingEventArgs> DownloadStarting;
+    }
+
+    runtimeclass CoreWebView2Download
+    {
+        // CoreWebView2Download
+        String Uri { get; };
+
+        String ContentDisposition { get; };
+
+        String MimeType { get; };
+
+        Int64 ExpectedDownloadSizeInBytes { get; };
```

Comment:
Don't need the "download" prefix; it's in the class name.

Reply:
Yes. See above.